lib/kafka/gzip_codec.rb

```diff
@@ -0,0 +1,28 @@
+module Kafka
+  class GzipCodec
```

Comment:
The codecs are defined by the Kafka spec – there's no way to build on top of this and stay compatible.